src/App.tsx

```diff
@@ -1,14 +1,51 @@
+/* eslint-disable no-console */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filter, setSelectedFilter] = useState('all');
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos().then(setAllTodos)
+
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const visibleTodos = allTodos
+    .filter((todo) => {
+      const trimmed = query.toLocaleLowerCase().trim();
```

Comment:
You should move this line to line 29 (Before the `const visibleTodos`). It is the same trimmed query, but it is calculated multiple times in the loop

Reply:
I understand! it is done! :D Thanks Umut